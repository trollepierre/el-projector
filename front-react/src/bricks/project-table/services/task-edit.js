const differ = () => {}
const remove = () => {}
const eliminate = () => {}


export {
  differ,
  remove,
  eliminate
}
