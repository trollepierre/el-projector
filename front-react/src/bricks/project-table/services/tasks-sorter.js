import moment from 'moment'

function sortByNext(project) {
  const sortByNext = (task1, task2) => {
    const moment1 = moment(task1.next, 'DD/MM/YYYY')
    const moment2 = moment(task2.next, 'DD/MM/YYYY')
    return moment1.unix() - moment2.unix()
  }
  return project.sort(sortByNext)
}

function sortByFirstName(project) {
  const sortByFirstName = (task1, task2) => {
    const name1 = `${task1.firstName} ${task1.lastName}`
    const name2 = `${task2.firstName} ${task2.lastName}`
    return name1 > name2 ? 1 : -1
  }
  return project.sort(sortByFirstName)
}

function sortByLastName(project) {
  const sortByLastName = (task1, task2) => {
    const name1 = `${task1.lastName} ${task1.firstName}`
    const name2 = `${task2.lastName} ${task2.firstName}`
    return name1 > name2 ? 1 : -1
  }
  return project.sort(sortByLastName)
}

function sortByPoints(project) {
  const sortByPoints = (task1, task2) => (task1.points < task2.points ? 1 : -1)
  return project.sort(sortByPoints)
}

function sortByLastDate(project) {
  const sortByLastDate = (task1, task2) => {
    const moment1 = moment(task1.lastDate, 'DD/MM/YYYY')
    const moment2 = moment(task2.lastDate, 'DD/MM/YYYY')
    return moment1.unix() - moment2.unix()
  }
  return project.sort(sortByLastDate)
}

function sortByOrigin(project) {
  const sortByOrigin = (task1, task2) => (task1.origin > task2.origin ? 1 : -1)
  return project.sort(sortByOrigin)
}

function sortByBirthday(project) {
  const sortBirthday = (task1, task2) => {
    if (!task1.nextBirthday) return 1
    if (!task2.nextBirthday) return -1
    const moment1 = moment(task1.nextBirthday, 'DD/MM/YYYY')
    const moment2 = moment(task2.nextBirthday, 'DD/MM/YYYY')
    return moment1.unix() - moment2.unix()
  }
  return project.sort(sortBirthday)
}

export {
  sortByBirthday,
  sortByFirstName,
  sortByLastDate,
  sortByLastName,
  sortByNext,
  sortByOrigin,
  sortByPoints,
}
