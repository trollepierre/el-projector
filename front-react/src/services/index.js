export { default as apiService } from './api'
export { default as tokenService } from './token'
export { default as loggerService } from './logger'
