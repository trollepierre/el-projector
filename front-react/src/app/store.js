import { withReducer } from '../store/withReducer';
import { apiService, tokenService } from '../services';

const SET_IS_AUTHENTICATED = 'app/SET_IS_AUTHENTICATED';

const setIsAuthenticated = isAuthenticated => dispatch => {
  if(!isAuthenticated){
    tokenService.removeTokens()
  }
  return dispatch({
    type: SET_IS_AUTHENTICATED, payload: { isAuthenticated }
  });
};

const authenticate = password => async dispatch => {
  try {
    const data = {
      email: 'some@example.org',
      name: password
    };
    const tokens = await apiService.post('login', data)
    await tokenService.saveUserTokens(tokens, data)
    dispatch({ type: SET_IS_AUTHENTICATED, payload: { isAuthenticated: true }})
  } catch (error) {
    dispatch({ type: SET_IS_AUTHENTICATED, payload: { isAuthenticated: false }})
    alert('wrong password')
  }
}

const loginSilently = () => async dispatch => {
  try {
    const refreshToken = tokenService.getRefreshToken();
    if(!refreshToken) return;
    const tokens = await apiService.post('login/token', { refreshToken })
    await tokenService.reauthenticate(tokens);
    await dispatch({ type: SET_IS_AUTHENTICATED, payload: { isAuthenticated: true }})
  } catch (error) {
    await dispatch({ type: SET_IS_AUTHENTICATED, payload: { isAuthenticated: false }})
    await tokenService.removeTokens()
    alert('problem during login silently')
    throw error
  }
}

export const actions = {
  setIsAuthenticated,
  authenticate,
  loginSilently,
};

export const initialState = {
  isAuthenticated: false,
};

export const actionHandlers = {
  [SET_IS_AUTHENTICATED]: (state, action) => {
    const { isAuthenticated } = action.payload;
    return {
      ...state,
      isAuthenticated,
    };
  },
};

export const reducer = withReducer(initialState, actionHandlers);
