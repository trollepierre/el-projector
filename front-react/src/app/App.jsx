import React from 'react';
import { Provider } from 'react-redux';
import ProjectPage from '../pages/ProjectPage/ProjectPage';
import { default as createStore } from '../store/createStore';
import { default as reducer } from '../bricks/project-table/project-body/store';
import LoginPage from '../pages/LoginPage';
import { auth } from '../services';

const store = createStore(reducer)

const App = () => {
  if(!auth.isAuthenticated()) return (<LoginPage/>)

  return (
    <div className="App">
      <Provider store={store}>
        <ProjectPage/>
      </Provider>
    </div>
  );
};

export default App;
