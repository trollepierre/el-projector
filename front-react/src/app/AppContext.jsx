import React, { useReducer, createContext, useContext } from 'react';
import PropTypes from 'prop-types';
import { reducer, initialState, actions } from './store';

const AppContext = createContext();

export const useAppContext = () => useContext(AppContext);

const getProps = (state, dispatch) => ({
  ...state,
  setIsAuthenticated: isAuthenticated => actions.setIsAuthenticated(isAuthenticated)(dispatch),
  authenticate: value => actions.authenticate(value)(dispatch),
});

export const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <AppContext.Provider value={{ ...getProps(state, dispatch) }}>
      {children}
    </AppContext.Provider>
  );
};

AppProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default AppContext;
