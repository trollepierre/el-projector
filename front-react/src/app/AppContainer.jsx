import React, { useEffect } from 'react';
import { AppProvider, useAppContext } from './AppContext';

import LoginPage from '../pages/LoginPage';
import { Header } from '../bricks';
import ProjectPage from '../pages/ProjectPage/ProjectPage';

const AppContainer = () => {
  const { isAuthenticated } = useAppContext();

  if (!isAuthenticated) {
    return (<LoginPage/>);
  }

  return (
    <div className="App">
      <Header/>
      <ProjectPage/>
    </div>
  );
};

const AppModule = () => {
  return (
    <AppProvider>
      <AppContainer/>
    </AppProvider>
  );
};

export default AppModule;
