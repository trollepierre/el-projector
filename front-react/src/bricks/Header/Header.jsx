import React from 'react';
import { Button } from '../../components';
import { tokenService } from '../../services';
import { useAppContext } from '../../app/AppContext';

const Header = () => {
  const { setIsAuthenticated } = useAppContext()

  const logout = () => {
    setIsAuthenticated(false)
    return tokenService.removeTokens();
  }

  return (
    <nav className="header">
      <Button onClick={logout} text='Logout'/>
    </nav>
  );
};

export default Header;
