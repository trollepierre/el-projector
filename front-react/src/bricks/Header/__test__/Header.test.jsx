import React from 'react';
import Header from '../Header';
import { shallow } from 'enzyme';
import * as AppContext from '../../../app/AppContext';
import { Button } from '../../../components';
import { tokenService } from '../../../services';

describe('Header', () => {
  let setIsAuthenticatedMock;

  beforeEach(() => {
    setIsAuthenticatedMock = jest.fn();
    jest
      .spyOn(AppContext, 'useAppContext')
      .mockImplementation(() => ({ setIsAuthenticated: setIsAuthenticatedMock }));
  });

  it('should match snapshot', () => {
    // When
    const wrapper = shallow(<Header/>);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  describe('events', () => {
    describe('onClick on logout button', () => {
      it('should set is authenticated to false', () => {
        // Given
        const wrapper = shallow(<Header/>);

        // When
        wrapper.find(Button).simulate('click');

        // Then
        expect(setIsAuthenticatedMock).toHaveBeenCalledWith(false);
      });

      it('should remove tokens', () => {
        // Given
        tokenService.removeTokens = jest.fn()
        const wrapper = shallow(<Header/>);

        // When
        wrapper.find(Button).simulate('click');

        // Then
        expect(tokenService.removeTokens).toHaveBeenCalledWith();
      });
    });
  });
});
