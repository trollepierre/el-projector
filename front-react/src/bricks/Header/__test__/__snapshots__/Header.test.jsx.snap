// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<nav
  className="header"
>
  <Button
    onClick={[Function]}
    text="Logout"
  />
</nav>
`;
